woocommerce-prevent-repeat-purchases.php

```diff
@@ -68,29 +71,53 @@ public function write_panel_save( $post_id ) {
 		update_post_meta( $post_id, 'prevent_repeat_purchase', empty( $_POST['prevent_repeat_purchase'] ) ? 'no' : 'yes' );
 	}
 
-    /**
-     * Scripts for the Admin to hide checkbox for products that aren't simple
-     */
-    public function admin_scripts() {
-        // Get Screens
-        $screen       = get_current_screen();
-        $screen_id    = $screen ? $screen->id : '';
-
-        // If this is the product edit screen
-        if ( in_array( $screen_id, [ 'product', 'edit-product' ] ) ) {
-            // JS to hide the option if it's not a simple product, just in case
-            wc_enqueue_js( "
-            jQuery( document.body ).on( 'woocommerce-product-type-change', function( event, value ) {
-                if ( value !== 'simple' ) {
-                    // Uncheck Checkbox
-                    jQuery( '#prevent_repeat_purchase' ).prop( 'checked', false );
-                    // Hide
-                    jQuery( '.prevent-repeat-purchase-wrap' ).hide();
-                }
-            });
-            " );
-        };
-    }
+	/**
+	 * Scripts for the Admin to hide checkbox for products that aren't simple
+	 */
+	public function admin_scripts() {
+		// Get Screens
+		$screen       = get_current_screen();
+		$screen_id    = $screen ? $screen->id : '';
+
+		// If this is the product edit screen
+		if ( in_array( $screen_id, [ 'product', 'edit-product' ] ) ) {
+			// JS to hide the option if it's not a simple product, just in case
+			wc_enqueue_js( "
+			jQuery( document.body ).on( 'woocommerce-product-type-change', function( event, value ) {
+				if ( value !== 'simple' ) {
+					// Uncheck Checkbox
+					jQuery( '#prevent_repeat_purchase' ).prop( 'checked', false );
+					// Hide
+					jQuery( '.prevent-repeat-purchase-wrap' ).hide();
+				}
+			});
+			" );
+		};
+	}
+
+	/**
+	 * Check to see if the product is purchasable
+	 */
+	public function is_product_repeat_purchasable( $product_id ) {
+		// Check for a value
+		$repeat_purchasable = get_transient( 'product_' . $product_id . '_repeat_purchaseable' );
+
+		// If there is no value in the transient... go get it
+		if ( $repeat_purchasable === false ) {
+			$repeat_purchasable_value = get_post_meta( $product_id, 'prevent_repeat_purchase', true );
+
+			// Set the transient for 5 minutes
+			set_transient( 'product_' . $product_id . '_repeat_purchaseable', $repeat_purchasable_value, 300 );
```

Comment:
Some logic to flush the 'product_' . $product_id . '_repeat_purchaseable' transient when you save the setting. Can be on the existing callback that saves the meta.